test/client/client.go

```diff
@@ -105,12 +108,59 @@ func makeHTTPRequest(client *http.Client) func(int) func() error {
 	}
 }
 
-func makeGRPCRequest(client pb.EchoTestServiceClient) func(int) func() error {
+func makeWebSocketRequest(client *websocket.Dialer) func(int) func() error {
 	return func(i int) func() error {
 		return func() error {
-			req := &pb.EchoRequest{Message: fmt.Sprintf("request #%d", i)}
+			req := make(http.Header)
+
+			log.Printf("[%d] Url=%s\n", i, url)
+			if headerKey == hostKey {
+				req.Add("Host", headerVal)
+				log.Printf("[%d] Host=%s\n", i, headerVal)
+			} else if headerKey != "" {
+				req.Add(headerKey, headerVal)
+				log.Printf("[%d] Header=%s:%s\n", i, headerKey, headerVal)
+			}
+
+			if msg != "" {
+				log.Printf("[%d] Body=%s\n", i, msg)
+			}
+
+			conn, _, err := client.Dial(url, req)
+			if err != nil {
+				// timeout or bad handshake
+				return err
+			}
+			// nolint: errcheck
+			defer conn.Close()
```

Comment:
does `// nolint: errcheck` have to be on the same line?

Reply:
err on.. ths worked..